core/templates/dev/head/components/SearchBarDirective.js

```diff
@@ -151,11 +167,68 @@ oppia.directive('searchBar', [function() {
             }
 
             _updateSelectionDetails('languageCodes');
+
+            if (Boolean(urlService.getUrlParams().q)) {
+              updateSearchFieldsBasedOnUrlQuery();
+            }
             _onSearchQueryChangeExec();
 
             _searchBarFullyLoaded = true;
           }
         );
+
+        var updateLanguageCodeFields = function(url) {
+          // Grabs Grab language code(s) from URL and returns url without
+          // language fields. This function assumes that language codes are
+          // at the end of the url.
+
+          var _languageCodePattern = / language_code=\([a-zOR\%\" ]+\)$/;
```

Comment:
Don't need all the underscores at the beginning -- we are moving away from that.